### PR TITLE
publish docker images upon successful ci master

### LINKED ISSUE #173: automate docker builds
should be able to automate `latest` and release images:
https://docs.docker.com/docker-hub/builds/

### PR DESCRIPTION
**Pull request by siggy**
fixes #173

### CODE REVIEW

**Comment by olix0r** [comment]
If we do this, we need to stop CI'ing jobs submitted from people outside of buoyant (otherwise, we expose our docker credentials to them...)

File: project/publish-docker.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+VERSION_TAG=$(docker images buoyantio/linkerd|sed -n 2p|awk '{print $2}')
+if [ -z "$VERSION_TAG" ]; then
+  echo "VERSION_TAG not found"
+  exit 1
+fi
+
+docker tag buoyantio/linkerd:$VERSION_TAG buoyantio/linkerd:latest
+docker tag buoyantio/namerd:$VERSION_TAG buoyantio/namerd:latest
+docker tag buoyantio/namerd:$VERSION_TAG-dcos buoyantio/namerd:latest-dcos
```
Comment (olix0r) [inline project/publish-docker.sh]: I don't think we should update latest to snapshot tags...

File: project/publish-docker.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+docker push buoyantio/linkerd:$VERSION_TAG
+docker push buoyantio/namerd:$VERSION_TAG
+docker push buoyantio/namerd:$VERSION_TAG-dcos
```
Comment (olix0r) [inline project/publish-docker.sh]: I think the sbt-docker plugin will do this for us.

**Comment by olix0r** [comment]
I don't think we need a separate script to do docker pushing.  I think we can simply use something like:
`./sbt linkerd/dockerBuildAndPush` or `./sbt linkerd/dockerPush`

**Comment by siggy** [comment]
Per offline discussion, disabling permissive forked builds. Forks can still ci, but won't have access to DOCKER_CREDENTIALS:
https://circleci.com/docs/fork-pr-builds/

**Comment by olix0r** [comment]
⭐  looks good!

**Pull request merged**